lib/rules/component-tags-order.js

```diff
@@ -156,18 +135,34 @@ module.exports = {
           return
         }
         const elements = getTopLevelHTMLElements()
+
+        const elementWithOrders = elements
+          .map((element) => {
+            const order = getOrderElement(element)
+            return {
+              order,
+              element
+            }
+          })
+          .filter(
+            /**
+             * @param { {order:OrderElement|undefined, element: VElement} } o
+             * @returns {o is {order:OrderElement, element: VElement}}
+             */
+            (o) => Boolean(o.order)
+          )
```

Comment:
Easier to read (IMHO) and saves one loop:

```suggestion
          .flatMap((element) => {
            const order = getOrderElement(element)
            return order ? [{ order, element }] : []
          })
```